// frontend/src/pages/Bookings.js
import React, { useEffect, useState } from "react";
import { Link, useNavigate } from "react-router-dom";
import API from "../services/api";

function Bookings() {
  const [bookings, setBookings] = useState([]);
  const [loading, setLoading] = useState(true);
  const navigate = useNavigate();
  const user = JSON.parse(localStorage.getItem("user"));

  // GET /api/bookings/user/:userId
  // Returns bookings with vehicleId populated (name, type, pricePerHour)
  useEffect(() => {
    if (!user) {
      navigate("/login");
      return;
    }
    API.get(`/bookings/user/${user._id}`)
      .then((res) => setBookings(res.data))
      .catch((err) => console.error(err))
      .finally(() => setLoading(false));
  }, []); // eslint-disable-line

  const totalSpent = bookings.reduce((sum, b) => sum + (b.totalPrice || 0), 0);

  return (
    <div
      style={{
        minHeight: "100vh",
        background: "var(--black)",
        paddingTop: "5rem",
      }}
    >
      {/* Header */}
      <div className="rent-header">
        <span className="section-label">Your History</span>
        <h1 className="section-title">MY BOOKINGS</h1>
        <p style={{ color: "var(--gray-400)", fontSize: "0.95rem", marginTop: "0.5rem" }}>
          Welcome back, {user?.name?.split(" ")[0]} 👋
        </p>
      </div>

      <div style={{ maxWidth: 900, margin: "0 auto", padding: "2rem 3rem" }}>
        {/* Stats row */}
        {!loading && bookings.length > 0 && (
          <div
            style={{
              display: "grid",
              gridTemplateColumns: "repeat(3, 1fr)",
              gap: "1rem",
              marginBottom: "2.5rem",
            }}
          >
            {[
              { label: "Total Bookings", value: bookings.length },
              { label: "Total Hours", value: bookings.reduce((s, b) => s + (b.hours || 0), 0) + "h" },
              { label: "Total Spent", value: "₹" + totalSpent },
            ].map((s) => (
              <div
                key={s.label}
                style={{
                  background: "var(--gray-800)",
                  border: "1px solid rgba(255,255,255,0.06)",
                  borderRadius: 16,
                  padding: "1.5rem",
                  textAlign: "center",
                }}
              >
                <div
                  style={{
                    fontFamily: "'Bebas Neue', sans-serif",
                    fontSize: "2rem",
                    letterSpacing: 1,
                    color: "var(--orange)",
                  }}
                >
                  {s.value}
                </div>
                <div style={{ fontSize: "0.78rem", color: "var(--gray-400)", textTransform: "uppercase", letterSpacing: 1, marginTop: 4 }}>
                  {s.label}
                </div>
              </div>
            ))}
          </div>
        )}

        {loading ? (
          <div style={{ textAlign: "center", padding: "4rem", color: "var(--gray-400)" }}>
            <div style={{ fontSize: "2rem", marginBottom: "1rem" }}>⏳</div>
            Loading your bookings...
          </div>
        ) : bookings.length === 0 ? (
          <div
            style={{
              textAlign: "center",
              padding: "4rem",
              background: "var(--gray-800)",
              borderRadius: 20,
              border: "1px solid rgba(255,255,255,0.06)",
            }}
          >
            <div style={{ fontSize: "3rem", marginBottom: "1rem" }}>🏍️</div>
            <h3 style={{ fontFamily: "'Bebas Neue', sans-serif", fontSize: "1.8rem", letterSpacing: 2, marginBottom: "0.5rem" }}>
              NO BOOKINGS YET
            </h3>
            <p style={{ color: "var(--gray-400)", marginBottom: "1.5rem" }}>
              You haven't rented anything yet. Time to hit the road!
            </p>
            <button className="btn btn-primary btn-big" onClick={() => navigate("/vehicles")}>
              Browse Vehicles
            </button>
          </div>
        ) : (
          <div style={{ display: "flex", flexDirection: "column", gap: "1rem" }}>
            {bookings.map((b, idx) => (
              <div
                key={b._id}
                style={{
                  background: "var(--gray-800)",
                  border: "1px solid rgba(255,255,255,0.06)",
                  borderRadius: 16,
                  padding: "1.5rem 2rem",
                  display: "grid",
                  gridTemplateColumns: "auto 1fr auto",
                  gap: "1.5rem",
                  alignItems: "center",
                  transition: "border-color 0.2s",
                }}
                onMouseEnter={(e) => e.currentTarget.style.borderColor = "rgba(232,79,29,0.3)"}
                onMouseLeave={(e) => e.currentTarget.style.borderColor = "rgba(255,255,255,0.06)"}
              >
                {/* Number badge */}
                <div
                  style={{
                    width: 44, height: 44,
                    background: "rgba(232,79,29,0.12)",
                    borderRadius: "50%",
                    display: "flex", alignItems: "center", justifyContent: "center",
                    fontFamily: "'Bebas Neue', sans-serif",
                    fontSize: "1.1rem",
                    color: "var(--orange)",
                    flexShrink: 0,
                  }}
                >
                  #{idx + 1}
                </div>

                {/* Info */}
                <div>
                  <div style={{ fontWeight: 600, fontSize: "1rem", marginBottom: 4 }}>
                    {b.vehicleId?.name || "Vehicle"}
                  </div>
                  <div style={{ display: "flex", gap: "1.5rem", flexWrap: "wrap" }}>
                    <span style={{ fontSize: "0.8rem", color: "var(--gray-400)" }}>
                      🏍️ {b.vehicleId?.type || "—"}
                    </span>
                    <span style={{ fontSize: "0.8rem", color: "var(--gray-400)" }}>
                      ⏱️ {b.hours} hour{b.hours > 1 ? "s" : ""}
                    </span>
                    <span style={{ fontSize: "0.8rem", color: "var(--gray-400)" }}>
                      📅 {b.bookingTime ? new Date(b.bookingTime).toLocaleDateString("en-IN", { day: "numeric", month: "short", year: "numeric", hour: "2-digit", minute: "2-digit" }) : "—"}
                    </span>
                  </div>
                </div>

                {/* Price */}
                <div style={{ textAlign: "right", flexShrink: 0 }}>
                  <div
                    style={{
                      fontFamily: "'Bebas Neue', sans-serif",
                      fontSize: "1.6rem",
                      letterSpacing: 1,
                      color: "var(--white)",
                    }}
                  >
                    ₹{b.totalPrice}
                  </div>
                  <div style={{ fontSize: "0.72rem", color: "var(--gray-400)", textTransform: "uppercase", letterSpacing: 1 }}>
                    Total paid
                  </div>
                </div>
              </div>
            ))}
          </div>
        )}

        {/* CTA */}
        {!loading && (
          <div style={{ textAlign: "center", marginTop: "2.5rem" }}>
            <button className="btn btn-primary btn-big" onClick={() => navigate("/vehicles")}>
              + Rent Another Vehicle
            </button>
          </div>
        )}
      </div>
    </div>
  );
}

export default Bookings;
